libs/libarchfpga/src/interposer_types.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+/**
+ * @file interposer_types.h
+ * @brief This file contains types used for parsing interposer-related tags such as <interposer_cut> and <interdie_wire>
+ * and converting that information into the device architecture-related data structures.
+ * 
+ */
+
+#include <unordered_map>
+#include <vector>
+#include <string>
+
+/**
+ * @brief Enum for direction of an interposer cut. X means horizontal cut and Y means vertical cut.
+ * 
+ */
+enum class e_interposer_cut_dim {
+    X,
+    Y
+};
+
+// Lookup table for converting between a character and an e_interposer_cut_dim
+inline const std::unordered_map<char, e_interposer_cut_dim> CHAR_INTERPOSER_DIM_MAP = {
+    {'X', e_interposer_cut_dim::X},
+    {'x', e_interposer_cut_dim::X},
+    {'Y', e_interposer_cut_dim::Y},
+    {'y', e_interposer_cut_dim::Y}};
+
+/**
+ * @brief Struct containing information of interdire wires i.e. connections between the dies on an interposer
+ * 
+ */
+struct t_interdie_wire_inf {
+    std::string sg_name;  ///< Name of the scatter-gather pattern to be used for the interdie connection
+    std::string sg_link;  ///< Name of the scatter-gather link to be used for the interdie connection
+    int offset_start;     ///< Starting point of scatter-gather instantiations
+    int offset_end;       ///< Ending point of scatter-gather instantiations
+    int offset_increment; ///< Increment/distance between scatter-gather instantiations
+    int num;              ///< Number of scatter-gather instantiations per switchblock location
+};
+
+/**
+ * @brief Struct containing information of an interposer cut
+ * 
+ */
+struct t_interposer_cut_inf {
+    e_interposer_cut_dim dim;                        ///< Dimension or axis of interposer cut
+    int loc;                                         ///< Location of the cut on the grid
```

Comment:
Give an example for this one.  If the interposer cut is at y = 13, is the cut above or below the tiles with coordinate y = 13?